src/pydata_sphinx_theme/__init__.py

```diff
@@ -131,19 +131,39 @@ def generate_nav_html(kind, startdepth=None, show_nav_level=1, **kwargs):
             for ul in soup("ul", recursive=False):
                 ul.attrs["class"] = ul.attrs.get("class", []) + ["nav", "bd-sidenav"]
 
+            # restructing html to handle part collapsing
+            partcaptions = soup.find_all("p", attrs={"class": "caption"})
+            if len(partcaptions):
+                new_soup = bs("<ul class='list-caption'></ul>", "html.parser")
+                for caption in partcaptions:
+                    for sibling in caption.next_siblings:
+                        if sibling.name == "ul":
+                            toclist = sibling
+                            break
+                    li = soup.new_tag("li", attrs={"class": "toctree-l0"})
+                    li.append(caption)
+                    li.append(toclist)
+                    new_soup.ul.append(li)
+            else:
+                new_soup = soup
+
             # Add icons and labels for collapsible nested sections
-            _add_collapse_checkboxes(soup)
+            _add_collapse_checkboxes(new_soup)
 
             # Open the navbar to the proper depth
-            for ii in range(int(show_nav_level)):
-                for checkbox in soup.select(
+            ii = 0
+            while True:
+                ii += 1
+                for checkbox in new_soup.select(
                     f"li.toctree-l{ii} > input.toctree-checkbox"
                 ):
                     checkbox.attrs["checked"] = None
```

Comment:
@AakashGfude I believe that the logic for automatically opening checkboxes for an active page is actually here:

https://github.com/pydata/pydata-sphinx-theme/blob/c98727693789c20a7790a3fa36029f5b5c9d6ed9/src/pydata_sphinx_theme/__init__.py#L295-L298

So you'd want to add a similar piece of logic to that block that also detects a caption that corresponds to that section of the toctree, and opens it.

Reply:
Thanks @choldgraf , I had to add the logic to expand the part explicitly here https://github.com/pydata/pydata-sphinx-theme/pull/594/commits/66982036bb6fe705ce10c19755076f759353799b#diff-d9f80958bbaaf03124bb877a07f31183c094105bbe78ea167de87e02b18d66b7R284, as it does not seem to have the `current` class when `show_nav_level`: 0. Not sure where is the logic to add `current` class to an element, but will try to find it.

Reply:
I believe that `current` is added by Sphinx, so perhaps after you've nested everything you can do something like 

```python
if new_soup.select(".current"):
  # Check the top `ul`
```

Reply:
@choldgraf, have implemented similar logic. if the child `li` elements have `current` class then "checking" the parent part